lib/hawkular/metrics/metric_api.rb

```diff
@@ -197,6 +222,9 @@ def initialize(client)
       # @return [Array[Hash]] rate points
       def get_rate(id, starts: nil, ends: nil, bucket_duration: nil)
         path = "/#{@resource}/#{ERB::Util.url_encode(id)}/rate"
+        if @new_api_supported
+          bucket_duration.nil? ? path << '/raw' : path << '/stats'
+        end
         params = { start: starts, end: ends, bucketDuration: bucket_duration }
```

Comment:
It doesn't matter because encode_params will eliminate parameters with nil values.